### PR TITLE
Added 'useTurnstile' hook

### PR DESCRIPTION
**Pull request by designly1**
Exposes the 'reset' method of the global turnstile object. It is often necessary to be able to programmatically reset a captcha. Example, when handling the error of an async fetch to process form data.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -43,6 +43,16 @@ let ensureTurnstile: () => Promise<any>;
...
 }
 
+export function useTurnstile() {
+  const reset = () => {
+    global.turnstile.reset();
+  }
+
+  return {
+    reset
+  }
```
Comment (Le0Developer) [inline src/index.tsx:53]: Sample implementation:
```suggestion
  const [state, setState] = useState(turnstileState);
  useEffect(() => {
    if(turnstileState === "ready") return;
    turnstileLoad.then(() => setState(turnstileState))
  }, []);
  return global.turnstile;
```

File: README.md [hunk 2]
```diff
@@ -11,10 +11,12 @@ npm i react-turnstile
 ## Usage
 
 ```jsx
-import Turnstile from "react-turnstile";
+import Turnstile, { useTurnstile } from "react-turnstile";
 
 // ...
 
+const { reset } = useTurnstile(); // Access turnstile reset method
```
Comment (Le0Developer) [inline README.md:18]: Please actually use the `reset()` function in the example, currently it's just a dead import.
Example: `onLoad={((widgetId) => setTimeout(reset, 30000, widgetId)) /* reset widget after 30s */}`

**Review by Le0Developer — CHANGES_REQUESTED** [review]
How exactly is this supposed to work when Turnstile is not yet loaded?
The api.js is only loaded once it's rendered, so scripts using this hook may try to access it before it's available and error.

Also this only exposes one out of 4 or so methods and is missing the optional argument for `turnstile.reset()`.

I think a better plan would be return `undefined` if turnstile is not yet loaded, and simply return the entire turnstile object once it is.

File: src/index.tsx (hunk 1 above)
Comment (designly1) [reply src/index.tsx:53]: Yeah, I thought about including the other methods. That's why I structured the code the way I did. Good point about checking if SDK is loaded. I'll modify and resubmit.

File: README.md (hunk 2 above)
Comment (designly1) [reply README.md:18]: Will do

File: src/index.tsx (hunk 1 above)
Comment (designly1) [reply src/index.tsx:53]: What is the 'state' for? It's not read anywhere in your example. Also I think an interval timer might be simpler to check when Turnstile is ready.

Comment (designly1) [reply src/index.tsx:53]: Actually, now that I think of it, if `global.turnstile` is undefined, `useTurnstile` will return undefined anyway. Why would I need additional logic?

File: README.md [hunk 3]
```diff
@@ -11,10 +11,20 @@ npm i react-turnstile
...
+// Programmatic access example:
+const turnstile = useTurnstile();
+
+const result = await fetch('/path/to/some/api');
+if (!result.ok) {
+  throw new Error(`Request failed with code ${result.status}`);
+  turnstile.reset();
+}
+
+
```
Comment (Le0Developer) [inline README.md:27]: Bad example, async components?
I'd move this into the onVerify callback

File: package-lock.json
Comment (Le0Developer) [inline package-lock.json:1]: The changes of this file don't belong here.

File: src/index.tsx (hunk 1 above)
Comment (Le0Developer) [reply src/index.tsx]: `state` is used to trigger a component re-render, so the application actually receives the turnstile object once it's loaded

**Comment by Le0Developer** [comment]
Implemented in 62ae777

**Pull request closed without merging**